vendor/plist_to_html/plist_to_html/PlistToHtml.py

```diff
@@ -0,0 +1,206 @@
+#!/usr/bin/env python
+
+import argparse
+import json
+import os
+import plistlib
+import shutil
+from xml.parsers.expat import ExpatError
+
+
+def get_file_content(filename):
+    with open(filename, 'r') as f:
+        return f.read()
+
+
+class HtmlBuilder:
+    """
+    Helper class to create html file from a report data.
+    """
+    def __init__(self, layout_dir):
+        self.layout_dir = layout_dir
+
+        self.header = get_file_content(
+            os.path.join(self.layout_dir, 'header.html'))
+        self.footer = get_file_content(
+            os.path.join(self.layout_dir, 'footer.html'))
+        self.style = get_file_content(
+            os.path.join(self.layout_dir, 'style.css'))
+        self.bug_viewer = get_file_content(
+            os.path.join(self.layout_dir, 'bugviewer.js'))
+
+        self.codemirror_css = get_file_content(
+            os.path.join(self.layout_dir, 'codemirror.min.css'))
+        self.codemirror_js = get_file_content(
+            os.path.join(self.layout_dir, 'codemirror.min.js'))
+        self.clike = get_file_content(
+            os.path.join(self.layout_dir, 'clike.min.js'))
+
+    def create(self, output_path, report_data):
+        """
+        Create html file with the given report data to the output path.
+        """
+        with open(output_path, 'w+') as html_output:
+            html_output.write(self.header)
```

Comment:
Are we sure we want to build the HTML like this? Wouldn't an extra layout file, which contains to-be-mapped replacings:

```html
<$HEADER$>
<style type="text/css">
  <$CODEMIRROR_CSS$>
  <!-- ... -->

  <$BUG_VIEWER$>
  var data = <$JSON_DATA$>;
</script>
<$FOOTER$>
```

and a representation of this "map" in the code be better? I know it would involve more `string.replace` operations, but would make general edits to the layout (which then would be an **actual layout** and not a dependency repository!) easier.

Reply:
It was a great idea :+1: